.github/workflows/publish.yml

```diff
@@ -0,0 +1,157 @@
+name: Publish
+
+env:
+  CLOUDSMITH_REPO: "ibm-dev"
+  RELEASE_VERSION: "1"
+
+on:
+  push:
+    branches:
+      - dev/publish-workflow-cloudsmith
```

Comment:
We should probably update this to `main` before merge.

Reply:
Also, do we want to publish a new package on every merge to main? I don't have any objections to that, but if we want to keep new releases more like barman's where we decide when new packages are going out, then we probably want to change this from an on push to `on: workflow_dispatch` which will give us a button to click in the GitHub Actions panel to run the workflow. [](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)

Reply:
Oh, I just reread the comment above about the `staging` vs `production` tags and processed it properly this time. Looks like it's all taken care of by that mechanism, so nevermind!

Reply:
Now you mention it this isn't quite analogous to Barman because we'd be overwriting the packages in the spooling repo each time we merge to `main`.

With Barman we use the git commit hash as the release/iteration version (unless it's a release build) so we don't have that problem - if we are going to publish every time we merge we should probably do similar here.

I think this is something we can address in the release process PR.